src/shared/RNGProvider.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useContext, useState } from 'react';
+
+import seedrandom, { PRNG } from 'seedrandom';
+
+export type RNGStore = Readonly<{
+  rng: PRNG;
+  setSeed(seed: string): PRNG;
+}>;
+
+export type RNGProviderProps = {
+  // Provide an initial seed so we guarantee `rng` isn't null
+  seed: string;
+  children: React.ReactNode;
+}
+
+const RNGContext = React.createContext<RNGStore>(null);
+export const RNGProvider: React.FC<RNGProviderProps> = ({ seed, children }) => {
+  // If we don't wrap `seedrandom` in a lambda like this, rng will be executed as state is set and it just becomes a number
+  const [ rng, setRng ] = useState<PRNG>(() => seedrandom(seed));
```

Comment:
Under this way of doing it does the seed not change? Just the rng function? or this is just the initial state with a passed in seed on launch and then the setSeed method below forces the recalculation?

Reply:
Initial state w/ passed in seed. `setSeed` below regenerates the rng function w/ a new seed.

Reply:
`seedrandom(seed)` generates a `PRNG` object which is a random number generator _function_. When this provider first loads, we provide it with a `seed` prop so `rng` is initialized immediately. We can change `rng` by using `setSeed` which will generate a new `PRNG` based on the updated seed we provide. This update is communicated to any components using `const { rng } = useRNG()` and can cause re-renders where `rng` is a dependency.

You can find an example of this behavior in `Sandbox` where I've added a button that will use `setSeed`. If you look at the console, you can see that `rng` is updated, and that causes the `useMemo` which calculates voronoi data to recompute since `rng` (a dependency to this memo) has changed.